config/routes.rb

```diff
@@ -1,3 +1,7 @@
 Spree::Core::Engine.routes.draw do
-  # Add your extension routes here
+  namespace :api, defaults: { format: 'json' } do
+    resource :affirm, only: [], controller: 'affirm' do
```

Comment:
What if people want to use their own controller at this route? What if they dont want the extension to be generating routes? Do we need to route in the spree namespace or can we define our own routes?